src/components/TaskDetail.jsx

```diff
@@ -55,7 +63,11 @@ export function TaskDetail({ task, isOpen, onClose, onUpdate, onDelete, onAddLab
   if (!isOpen || !task) return null;
 
   const handleSave = () => {
-    onUpdate(task.id, { description });
+    onUpdate(task.id, {
+      description,
+      priority,
+      dueDate: dueDate?.toISOString(),
+    });
     setIsEditing(false);
   };
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
const handleSave = async () => {
    try {
      setIsEditing(false);
      await onUpdate(task.id, {
        description,
        priority,
        dueDate: dueDate?.toISOString(),
      });
    } catch (error) {
      setIsEditing(true);
      console.error('Failed to update task:', error);
      // Handle error appropriately (e.g., show error message)
    }
  };
```
The handleSave function directly triggers a state update and API call without handling potential errors or loading states. Consider adding error handling and loading state management.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;